src/circuit/CircuitKernel.java

```diff
@@ -0,0 +1,37 @@
+package circuit;
+
+import components.standard.Standard;
+
+/**
+ * Circuit kernel component with primary methods.
+ */
+public interface CircuitKernel extends Standard<Circuit> {
+
+    /**
+     * Reports total {@Code this} current.
+     *
+     * @return the current value
+     * @ensures {@Code this can be used to obtain current}
+     */
+    double getTotalCurrent();
+
+    /**
+     * Reports added {@Code this} resistance.
+     *
+     * @return total resistance in series.
+     * @ensures {@Code this can be used to find potenital resistance}
+     */
+    double getTotalResistance();
```

Comment:
How are these things set? Are they set in the constructor? In that case, are circuits immutable?